api/v1alpha4/openstackcluster_types.go

```diff
@@ -58,14 +58,29 @@ type OpenStackClusterSpec struct {
 	ExternalNetworkID string `json:"externalNetworkId,omitempty"`
 
 	// ManagedAPIServerLoadBalancer defines whether a LoadBalancer for the
-	// APIServer should be created. If set to true the following properties are
-	// mandatory: APIServerFloatingIP, APIServerPort
+	// APIServer should be created.
 	// +optional
 	ManagedAPIServerLoadBalancer bool `json:"managedAPIServerLoadBalancer"`
 
-	// APIServerFloatingIP is the floatingIP which will be associated
-	// to the APIServer. The floatingIP will be created if it not
-	// already exists.
+	// DisableAPIServerFloatingIP determines whether or not to attempt to attach a floating
+	// IP to the API server. This allows for the creation of clusters when attaching a floating
+	// IP to the API server (and hence, in many cases, exposing the API server to the internet)
+	// is not possible or desirable, e.g. if using a shared VLAN for communication between
+	// management and workload clusters or when the management cluster is inside the
+	// project network.
+	// This option requires that the API server use a VIP on the cluster network so that the
+	// underlying machines can change without changing ControlPlaneEndpoint.Host.
+	// When using a managed load balancer, this VIP will be managed automatically.
+	// If not using a managed load balancer, cluster configuration will fail without additional
+	// configuration to manage the VIP on the control plane machines, which falls outside of
+	// the scope of this controller.
+	// +optional
+	DisableAPIServerFloatingIP bool `json:"disableAPIServerFloatingIP"`
```

Comment:
sorry, is the no floating ip means only fixed ip, no floating ip in tenant network or it must be using provider network?

Reply:
It literally does what it says on the tin and disables the floating IP for the API server. It makes no other assumptions about the networking.

However the cluster will only create successfully when the management cluster and workload cluster are on the same network, which could be a tenant-specific VXLAN or a shared provider VLAN.

That means that it is possible to create a workload cluster whose API server is not accessible, in which case the cluster creation will fail. For example, if you set this flag when using the automatically created cluster network, then the API server load balancer will only exist on the newly created network and will not be accessible by the management cluster.

Reply:
What we can't support without a fair amount of extra work, but would be nice to support in the future, is the case where you have two networks - a workload-cluster-specific VXLAN and a VLAN provider "control network".

In this case, the management cluster lives on the "control network", and the worker nodes for the workload cluster live on the cluster-specific VXLAN. The control-plane nodes for the workload cluster are connected to both, with the management cluster accessing the API server via the "control network" (without a floating IP), and the worker nodes accessing the API server via the cluster-specific network.

In order to do this, we would need to make a load balancer for the API *on each network*, and also be able to tell the corresponding `kubeadmcontrolplane` to include IP SANs for each IP in the TLS certificate for the API server. Ideally this wouldn't require any configuration from the user as the IPs for the load balancers won't be known until they are deployed.

Reply:
ok , I admin I am not fully understand the whole topo here but I think it's anyway a future item
maybe create another issue with `topo picture` will be easy to understand the desired way

for now, with following statement I think we are good enough for this PR.. thanks 

>>>will only create successfully when the management cluster and workload cluster are on the same network